src/core/Clip.cpp

```diff
@@ -116,7 +116,7 @@ void Clip::movePosition( const TimePos & pos )
 void Clip::changeLength( const TimePos & length )
 {
 	m_length = length;
-	Engine::getSong()->updateLength();
+	if (getTrack()) { getTrack()->updateLength(); }
```

Comment:
I'm tempted to say let's remove this `if` and just call directly. We shouldn't ever have an instance of a clip without a track that I can think of.

Reply:
That's what I thought, but placing a note inside of a new clip consistently crashed LMMS, so it's clearly needed.

Reply:
Do you have a backtrace for the crash? I can try myself if you don't.

Reply:
Ah, let's make a note of that in a comment then.
```cpp
// adding a note to a new clip will crash LMMS without this if check
// TODO: periodically check if the if is needed (a clip should always be connected to a track)
```

Reply:
Or @PhysSong will find the problem in a bit 😉
It's probably something creating a clip without a track somewhere.

Reply:
The traceback leads back to the Track's updateLength function, at `for (const auto& clip : m_clips)`.  There are several other parts in Clip.cpp that check for the Track's existence before doing things already.